Solutions/STM32F4DISCOVERY/scatterfile_gcc_missing_symbols.xml

```diff
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <ScatterFile xmlns="http://schemas.microsoft.com/netmf/ScatterfileSchema.xsd">
+  <!-- 
+    - using newlib-nano adds libnosys.a, which calls sbrk, which requires the 'end' symbol
+    - this is pulled in by a _printf imported in function _Crypto_StepRSAOperation @ Crypto.lib(rsa_zenith.obj) 
+  -->
+  <Provide Name="end" />

```

Comment:
That seems to be the cause. If you build the tinybooter with GCC you'll see the error with the description going down all the way to a printf in a lib from crypto.
I'm 100% with you that this crypto stuff should be replaced for good.